public/js/addsales.js

```diff
@@ -0,0 +1,63 @@
+
+function init() {
+    var itemID = document.getElementById('ItemId');
+    var itemName = document.getElementById('ItemName');
+    var price = document.getElementById("Price");
+    var qty = document.getElementById("Quantity");
+    var date = document.getElementById("Date");
+    var statetxt = document.getElementById("statetxt");
+    var addSalesForm = document.getElementById("addSalesForm");
+    
+    
+    var idPattern = /^[a-zA-Z]+$/;
+    var namePattern = /^[A-Za-z ]+/;
+    var numberPattern = /^[0-9]+]/;
+    itemName.onkeypress = function() {  
+        if (!itemName.value.match(namePattern)) {
+            statetxt.innerHTML = "Item name can contain only letters and whitespaces!";
+            statetxt.style.color = "red";
+            statetxt.style.display = "block";
+        } else {
+            statetxt.innerHTML = "";
+        }
+
+    }
+    itemID.onkeypress = function() {
+        if (!itemID.value.match(idPattern)) {
+            statetxt.innerHTML = "Item id can contain only letters!";
+            statetxt.style.color = "red";
+            statetxt.style.display = "block";
+        } else {
+            statetxt.innerHTML = "";
+        }
+    }
+    price.onkeypress = function() {
+        if (!price.value.match(numberPattern)) {
```

Comment:
What if the price of a particular item is $4.99 or any float value? In my view, the price of the item can be either float (2 decimal places) or int value.

Reply:
again, I used the same pattern used for price in addsales.php